dart/dynamics/Frame.cpp

```diff
@@ -0,0 +1,478 @@
+{
+  if(mAmWorld)
+    return mAcceleration;
+
+  if(mNeedAccelerationUpdate)
+  {
+    mAcceleration = math::AdInvT(getRelativeTransform(),
+                                 getParentFrame()->getSpatialAcceleration())
+        + getPrimaryRelativeAcceleration()
+        + getPartialAcceleration();
+//        + getRelativeSpatialAcceleration()
+//        + math::ad(getSpatialVelocity(), getRelativeSpatialVelocity());
+
+    mNeedAccelerationUpdate = false;
+  }
+
+  return mAcceleration;
+}
+
+//==============================================================================
+Eigen::Vector6d Frame::getSpatialAcceleration(
+    const Frame* _inCoordinatesOf) const
+{
+  if(this==_inCoordinatesOf)
+    return getSpatialAcceleration();
+
+  if(_inCoordinatesOf->isWorld())
+    return math::AdR(getWorldTransform(), getSpatialAcceleration());
+
+  return math::AdR(getTransform(_inCoordinatesOf), getSpatialAcceleration());
+}
+
+//==============================================================================
+Eigen::Vector6d Frame::getSpatialAcceleration(
+    const Frame* _relativeTo, const Frame* _inCoordinatesOf) const
+{
+  // Frame 2: this, Frame 1: _relativeTo, Frame O: _inCoordinatesOf
+  // Acceleration of Frame 2 relative to Frame 1 in coordinates of O: a_21[O]
+  // Total acceleration of Frame 2 in coordinates of Frame 2: a_2[2]
+  // Velocity of Frame 2 relative to Frame 1 in coordinates of Frame 2: v_21[2]
+
+  // a_21[O] = R_O2*( a_2[2] - X_21*a_1[1] - v_2[2] x v_21[2] )
+
+  return math::AdR(getTransform(_inCoordinatesOf),
+                   getSpatialAcceleration()
+                   - math::AdT(_relativeTo->getTransform(this),
+                               _relativeTo->getSpatialAcceleration())
+                   - math::ad(getSpatialVelocity(),
+                              getSpatialVelocity(_relativeTo,this)));
+}
+
+//==============================================================================
+Eigen::Vector3d Frame::getLinearAcceleration(
+    const Frame* _relativeTo, const Frame* _inCoordinatesOf) const
+{
+  const Eigen::Vector6d& v_rel = getSpatialVelocity(_relativeTo,
+                                                    _inCoordinatesOf);
+
+  // r'' = a + w x v
+  return getSpatialAcceleration(_relativeTo,_inCoordinatesOf).tail<3>()
+         + v_rel.head<3>().cross(v_rel.tail<3>());
+}
+
+//==============================================================================
+Eigen::Vector3d Frame::getAngularAcceleration(
+    const Frame* _relativeTo, const Frame* _inCoordinatesOf) const
+{
+  return getSpatialAcceleration(_relativeTo, _inCoordinatesOf).head<3>();
+}
+
+//==============================================================================
+template <typename T>
+static std::set<const T*> convertToConstSet(const std::set<T*>& _set)
+{
+  std::set<const T*> const_set;
+  typename std::set<T*>::const_iterator it=_set.begin(), end=_set.end();
+  for( ; it != end; ++it)
+    const_set.insert(*it);
```

Comment:
Since we support C++11, I advocate to use range-based for loop like:
```cpp
  for (const auto& element : _set)
    const_set.insert(element);
```
It makes more readable.